flat_kivy_extensions/flatappextension.py

```diff
@@ -0,0 +1,366 @@
+
+from kivy.uix.widget import Widget
+from kivy.lang import Builder
+from kivy.uix.boxlayout import BoxLayout
+from kivy.uix.screenmanager import Screen, FadeTransition, NoTransition
+
+from flat_kivy.flatapp import FlatApp
+from flat_kivy.uix.flattextinput import FlatTextInput
+from flat_kivy.uix.flaticonbutton import FlatIconButtonLeft
+from flat_kivy.uix.flatlabel import FlatLabel
+from flat_kivy.font_definitions import style_manager
+
+from flat_kivy_extensions.uix.customiconbutton import CustomIconButton
+
+Builder.load_string('''
+#:import NavigationDrawer kivy_garden.navigationdrawer.NavigationDrawer
+#:import NoTransition kivy.uix.screenmanager.NoTransition
+#:import FadeTransition kivy.uix.screenmanager.FadeTransition
+<RootWidget>:
+    title: 'Some App Title'
+    background_color: 1.0, 1.0, 1.0
+    header_height: dp(40)
+    header_color: 0.1, 0.3, 0.2
+    side_panel_color: 0.1, 0.1, 0.1
+
+    BoxLayout:
+        orientation: 'vertical'
+        size_hint: (None, None)
+        size: root.size
+        canvas.before:
+            Color:
+                rgba: root.background_color
+            Rectangle:
+                pos: self.pos
+                size: self.size
+
+        HeaderLayout:
+            id: header
+            size_hint_y: None
+            height: root.header_height
+            title: root.title
+            color: root.header_color
+            header_color: root.header_color
+
+        NavigationDrawer:
+            id: navigationdrawer
+            height: root.height
+            width: root.width
+            side_panel_width: min(dp(250), 0.6*self.width)
+            anim_type:  'slide_above_simple'
+
+            StackLayout:
+                orientation: 'tb-lr'
+                id: side_panel
+                #orientation: 'vertical'
+                padding: '3dp'
+                spacing: '3dp'
+
+                # this will be populated dynamically when the root application is built
+
+            BoxLayout:
+                orientation: 'vertical'
+
+                canvas.before:
+                    Color:
+                        rgb: root.background_color
+                    Rectangle:
+                        pos: self.pos
+                        size: self.size
+
+                ScreenManager:
+                    id: screenmanager
+                    transition: NoTransition()
+                    # Question: When selecting FadeTransition the switching
+                    #           annimation seems to go black when switching,
+                    #           how can this be avoided?
+                    transition: FadeTransition()
```

Comment:
Do we have a button to cause a transition yet? Wanted to test this issue myself but I can't seem to figure out how to change screens to trigger it.

Reply:
Sure - any button on the navigation slide out panel should force a screen switch with the specified transition.  Does this not work for you?

Reply:
Hrmm I only get one button in the navigation slide out panel, should there be more?

Reply:
Ahh - okay, I think the reason you are not seeing this is because the demo.py file in the master branch just puts one screen into the screen manager and does not do the auto-population of the buttons in the nav panel that link to a screen.

Do you recall which branch/commit version you were testing with?

If you try the source from the head of the application-scaffolding branch you can see that in the demo.py file there are multiple screens included and the corresponding navigation buttons are included in flatappextension in the build() step.

If the head of the application-scaffolding does not work for you just let me know.

Reply:
Ahh - okay, I think the reason you are not seeing this is because the demo.py file in the master branch just puts one screen into the screen manager and does not do the auto-population of the buttons in the nav panel that link to a screen.

Do you recall which branch/commit version you were testing with?

If you try the source from the head of the application-scaffolding branch you can see that in the demo.py file there are multiple screens included and the corresponding navigation buttons are included in flatappextension in the build() step.

If the head of the application-scaffolding does not work for you just let me know.